pkgs/development/python-modules/austin-python/default.nix

```
@@ -0,0 +1,88 @@
+{ stdenv
+, lib
+, buildPythonPackage
+
+, austin ? null  # Avoid circular dependency between austin / austin-python.
+, fetchFromGitHub
+, hatchling
+, hatch-vcs
+, poetry-core
+, protobuf
+, psutil
+, pytestCheckHook
+, pythonOlder
+, pythonRelaxDepsHook
+, toml
+}:
+
+buildPythonPackage rec {
+  pname = "austin-python";
+  version = "1.4.2";
+  format = "pyproject";
+
+  disabled = pythonOlder "3.7";
+
+  src = fetchFromGitHub {
+    owner = "p403n1x87";
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256-ddzv17lxj+59NMEe13eUNUmuzezrLY5q08+2Rrocp9E=";
+  };
+
+  postPatch = ''
+    # Fixup paths to dependencies usually found in PATH.
+    substituteInPlace austin/tools/resolve.py \
+      --replace '"addr2line"' '"${stdenv.cc.bintools}/bin/addr2line"'
+    ${if austin == null then ""
+      else ''
+        substituteInPlace austin/__init__.py \
+          --replace 'return self.BINARY' 'return "${austin}/bin/austin"'
+      ''}
+  '';
+
+  SETUPTOOLS_SCM_PRETEND_VERSION = version;
+
+  nativeBuildInputs = [
+    hatchling
+    hatch-vcs
+    poetry-core
+    pythonRelaxDepsHook
+  ];
+
+  propagatedBuildInputs = [
+    protobuf
+    psutil
+    toml
+  ];
+
+  checkInputs = [
+    pytestCheckHook
+  ];
+
+  doCheck = austin != null;
```

```suggestion
  doCheck = false;
```

we must use something like https://github.com/NixOS/nixpkgs/blob/c4c1866143ec02016d30eabed22cd6d0de39a10b/pkgs/development/python-modules/automat/default.nix#L50

---

(Let's dedup with the thread about $PATH lookup since this only exists as a direct consequence to it.)

---

We can escape the infinity recursion here by disabling checks instead of overwriting a package to null. This also removes the antipattern and things like austin.version do not break.

---

I don't see how this is supposed to work:

- austin requires austin-python for its checkPhase
- austin-python needs to be patched to point to austin for it to be freestanding in nixpkgs

If you just make austin depend on austin-python minus its own tests, there's still a circular dependency because of the reference in postPatch.

---

We can break this by only running tests for austin in its passthru.